src/hex.cpp

```diff
@@ -86,11 +86,8 @@ namespace libtorrent
 	std::string to_hex(span<char const> in)
 	{
 		std::string ret;
-		if (!in.empty())
-		{
-			ret.resize(in.size() * 2);
-			to_hex(in.data(), in.size(), &ret[0]);
-		}
+		ret.resize(in.size() * 2);
+		to_hex(in.data(), in.size(), &ret[0]);
```

Comment:
this still potentially calls ``operator[]`` on an empty string, which is not allowed. you have to call ``ret.data()``